src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -350,7 +407,16 @@ private void RunOnce(bool checkConditions)
                     }
                 }
 
-                // TODO(cretz): Check conditions
+                // Collect all condition sources to mark complete and then complete them. This
+                // avoids modify-during-iterate issues.
+                if (checkConditions)
```

Comment:
This is better than what we do in Python where we check conditions only after each batch of jobs.

I'd have to test but with this approach it seems like we're less likely to reach `A-B-A` situations where the condition reaches `B` and is immediately flipped back to `A` (e.g. if set by different signal handlers).

Reply:
This checks after each batch kinda too. Note the nested loops.

Reply:
We trigger `RunOnce` after applying every job, not once per batch.

Reply:
Which is the right behavior AFAICT but I have not confirmed there's an issue with the way it's done in Python.
In any case it's an edge case.

Reply:
Ah, yes, that is correct I forgot that I don't group jobs I just order them. I agree that batch + run once is a bit more confusing.